packages/core/src/messagingProtocols/mls/E2EIdentityService/E2EIServiceExternal.ts

```diff
@@ -81,7 +81,13 @@ export class E2EIServiceExternal extends TypedEventEmitter<Events> {
     return this.coreCryptoClient.e2eiIsEnabled(this.mlsService.config.cipherSuite);
   }
 
-  public async getAllGroupUsersIdentities(groupId: string): Promise<Map<string, DeviceIdentity[]>> {
+  public async getAllGroupUsersIdentities(groupId: string): Promise<Map<string, DeviceIdentity[]> | undefined> {
+    const conversationExists = await this.mlsService.conversationExists(groupId);
+
+    if (!conversationExists) {
+      return undefined;
+    }
+
```

Comment:
do we really need the check here since `getUsersIdentities` also perform this check?

Reply:
`mlsService.getClientIds` on a line below would throw as well if the mls group does not exist

Reply:
oh I see, makes sense :)